lms/djangoapps/courseware/courses.py

```diff
@@ -462,24 +461,24 @@ def get_course_date_blocks(course, user, request=None, include_access=False,
     sorted by date.
     """
     block_classes = [
+        CertificateAvailableDate,
         CourseEndDate,
+        CourseExpiredDate,
         CourseStartDate,
         TodaysDate,
         VerificationDeadlineDate,
         VerifiedUpgradeDeadlineDate,
     ]
-    if not course.self_paced and certs_api.get_active_web_certificate(course):
-        block_classes.insert(0, CertificateAvailableDate)
 
     blocks = [cls(course, user) for cls in block_classes]
     if RELATIVE_DATES_FLAG.is_enabled(course.id):
-        blocks.append(CourseExpiredDate(course, user))
         blocks.extend(get_course_assignment_date_blocks(
             course, user, request, num_return=num_assignments,
             include_access=include_access, include_past_dates=include_past_dates,
         ))
```

Comment:
I chose to leave these under the `if` block (rather than adding an `CourseAssignmentDate.is_allowed` property) becase this method actually does some work, which we probably want to avoid outside of PLS.